charts/consul/templates/server-config-configmap.yaml

```diff
@@ -72,6 +72,9 @@ data:
               "ca_file": "/consul/vault-ca/tls.crt",
               {{- end }}
               "intermediate_pki_path": "{{ .connectCA.intermediatePKIPath }}",
+              {{- if .vaultNamespace }}
+              "namespace": "{{ .vaultNamespace }}",
+              {{- end }}
```

Comment:
I think we would want to check `additionalConfig` similar to the annotations so we don't duplicate namespace values.

I think I figured out how to do actual key checks rather than `contains` in the embedded JSON and YAML, [here's](https://helm-playground.com/#t=LQhQEMDtIewF3HAljSBnASgUzQVwDZwBcoA3qcAARIBmlAFFACYMB0AauPrjqwOb4YAIy6s0WAMYAnLHDQAhcBIDWWSE1YA3cAThadhAHLgAtjgAOSrAEoGsOAwAW4NAGksATwZMsNAw4AiAMoOLh40fkERfDFJGTlFFTUNbV1WcD41OABBaHhEFHRKAB9KGikYEwBNU3xbANTCVmc0RyQJGClzVg6TAHpIUwsrAOsxgF9x0EpKBv9mlzaOrp7KgaG0SwksAKJZ8hDObl4BYVFxaVkFJVV1fTTGuGMzTatKSYCyCkpk96nQDqQGhIPjYPCEEgzchUADuSDgjkOYROUXOcSuiVuKX8f2mlEBkEkcGAEnAJNQwL4rAAVmhUHtinioUyZrMCUTdpQANos1mUUi8vls8AAfUBlM5PKF0v5gpls3ATCYMjQaE5AQOq2gRIAwtl0kqVWg-gEADRymXQ6h0ZghUmxS5PIZ28AO%2BLuLyTC3SgKkkXA-A7PYBPqA8F9R4k8B9OD4CLSOBm71Cq2-L3yn1ISBwLBSMxMJCILAi8zKJAlxCOdWa9kSOB61hZnN5rAFosABVcAEl25WTeaMynvrQGLb6PcjBstjY7PAnC4Pd5fDignbUIS6w3FQXkKguDqKSCSmUKiYAFJ0yD1QYvaejMa4wd8gI34bbaukEKPZ5vrD95N8qm6iPk%2BMwBBU8AlmWFYIh%2Ba7apu%2BoQXAnY9n2HwDqBCq4AiIpmAiMBMJyApYayARwB45hBrMyi4EIuaEjmaqYaRAQmDAuDZjBVbBjW666vqOgIgAsrIjiEb2CL-qRYGWFIphqnsJEyWBFSBnBWrggAyrmmi5hgMCBiaAFClMpFmU%2BFnygAuixMq%2BqK5gVJoSA%2BFI6qPJ88pWay1kshZQEsJMQA&v=LQhQHMBsHsCMENIC5QAJUGcCmBjATlgC4YBC8OA1lgHYAmK66AbvAK6SEOPNscBy8ALZYMAB3JYkqAESFoo4JCxMskYNQzS03VPHA1CAQWrVoheIQCW0DVIA%2B2nahbtCAOgAW8DB8s5oeKJu-oIA9NRCIuI4kqgAZtDQwPAmZhbW1I46Lhye3r7%2BgcHQYWYeWHhSCHhZ6P4muIQAwoZcTvC0tJZWNohNNnGW4Pa1OgDeqKNOMhHCYhLSUtIJSQBWGDbSADRTTtJlFYsy1VrTjAC%2BoEA) an example.

Something like that should allow us to have clean mutual exclusion on both config entries, w/ the more specific values taking precedence over `vault.vaultNamespace`, I think. In this case, it would look like:

```suggestion
              {{- if (and (.vaultNamespace) (not (hasKey (default "" .connectCA.additionalConfig | fromJson) "namespace"))) }}
              "namespace": "{{ .vaultNamespace }}",
              {{- end }}
```

For the annotation checks, I think we can do:
```
{{- if (and (.Values.global.secretsBackend.vault.vaultNamespace) (not (hasKey (default "" .Values.global.secretsBackend.vault.agentAnnotations | fromYaml) "vault.hashicorp.com/namespace")))}}
  "vault.hashicorp.com/namespace": "{{ .Values.global.secretsBackend.vault.vaultNamespace }}"
{{- end }}
```